geoutils/geovector.py

```diff
@@ -68,64 +77,67 @@ def bounds(self) -> rio.coords.BoundingBox:
         """Get a bounding box of the total bounds of the Vector."""
         return rio.coords.BoundingBox(*self.ds.total_bounds)
 
-    def copy(self):
+    def copy(self: VectorType) -> VectorType:
         """Return a copy of the Vector."""
         # Utilise the copy method of GeoPandas
-        return Vector(self.ds.copy())
+        new_vector = self.__new__(type(self))
+        new_vector.__init__(self.ds.copy())
+        return new_vector  # type: ignore
 
-    def crop2raster(self, rst):
+    def crop2raster(self, rst: gu.Raster) -> None:
         """
-        Update self so that features outside the extent of a raster file are cropped. Reprojection is done on the fly if both data set have different projections.
+        Update self so that features outside the extent of a raster file are cropped.
+
+        Reprojection is done on the fly if both data set have different projections.
 
         :param rst: A Raster object or string to filename
-        :type rst: Raster object or str
         """
         # If input is string, open as Raster
         if isinstance(rst, str):
-            from geoutils.georaster import Raster
-            rst = Raster(rst)
+            rst = gu.Raster(rst)
 
         # Convert raster extent into self CRS
         # Note: could skip this if we could test if rojections are same
         # Note: should include a method in Raster to get extent in other projections, not only using corners
         left, bottom, right, top = rst.bounds
-        x1, y1, x2, y2 = warp.transform_bounds(
-            rst.crs, self.ds.crs, left, bottom, right, top)
+        x1, y1, x2, y2 = warp.transform_bounds(rst.crs, self.ds.crs, left, bottom, right, top)
         self.ds = self.ds.cx[x1:x2, y1:y2]
 
-    def create_mask(self, rst=None, crs=None, xres=None, yres=None, bounds=None):
+    def create_mask(
+        self,
+        rst: str | gu.georaster.RasterType | None = None,
```

Comment:
Is this format preferred over using Union?

Reply:
It's something that `pyupgrade` did automatically. I do think it's cleaner though, after getting used to it!